### PR TITLE
Support Node 4 again

### PR DESCRIPTION
**Pull request by sommestad**
Some projects still need Node 4-compatibility.

### CODE REVIEW

**Comment by sommestad** [comment]
@dbrockman Please verify that the file setup for NPM publishing is correct.

File: .npmignore [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# IntelliJ
+.idea
+*.iml
+
+coverage
+
+yarn.lock
+.eslintcache
+
+lib
```
Comment (dbrockman) [inline .npmignore]: - [ ] use the `files` field in package.json instead of a `.npmignore` file

Comment (dbrockman) [reply .npmignore]: In package.json:

```json
{
  ...
  "main": "lib/client-factory.js",
  "files": [
    "lib"
  ],
  ...
}
```

Comment (sommestad) [reply .npmignore]: There's already:

```
  "files": [
    "build"
  ]
```

Does that mean I can remove the .npmignore altogether?

File: package.json [hunk 2]
```diff
@@ -54,10 +59,13 @@
...
-    "node": ">=8",
+    "node": ">=4",
     "npm": ">=5"
   },
   "release": {
     "analyzeCommits": "simple-commit-message"
-  }
+  },
+  "files": [
+    "build"
```
Comment (dbrockman) [inline package.json]: ok, I didn't see that you added `files`. In that case you can just remove the `.npmignore` file

File: package.json [hunk 3]
```diff
@@ -3,12 +3,14 @@
   "description": "Module for sending requests to other APIs",
   "main": "lib/client-factory.js",
```
Comment (dbrockman) [inline package.json]: - [ ] change this to "build"

```diff
- "main": "lib/client-factory.js",
+ "main": "build/client-factory.js",
```

File: .npmignore (hunk 1 above)
Comment (dbrockman) [reply .npmignore]: yes

Comment (dbrockman) [reply .npmignore]: also don't forget to update the main field

File: package.json [hunk 4]
```diff
@@ -20,15 +22,18 @@
...
-    "eslint-plugin-import": "^2.2.0",
-    "eslint-plugin-mocha": "^4.8.0",
+    "eslint": "^4.7.2",
+    "eslint-config-springworks": "^7.0.11",
+    "eslint-plugin-import": "^2.7.0",
+    "eslint-plugin-mocha": "^4.11.0",
     "eslint-plugin-should-promised": "^2.0.0",
-    "eslint-plugin-springworks": "^2.0.1",
+    "eslint-plugin-springworks": "^2.1.0",
     "greenkeeper-postpublish": "^1.0.1",
```
Comment (dbrockman) [inline package.json]: can you remove greenkeeper-postpublish in this PR? and I'll delete mine if it's still open

**Comment by sommestad** [comment]
@dbrockman if there's some Babel magic you know to get this to work with babel-env, I'm all ears. Almost tore my hair out to get this to work properly...

**Pull request merged**